src/main/java/com/angelbroking/smartapi/http/SmartAPIResponseHandler.java

```diff
@@ -14,25 +9,35 @@
 import com.angelbroking.smartapi.http.exceptions.PermissionException;
 import com.angelbroking.smartapi.http.exceptions.SmartAPIException;
 import com.angelbroking.smartapi.http.exceptions.TokenException;
-
+import com.angelbroking.smartapi.models.SearchScripResponseDTO;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
 import okhttp3.Response;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.List;
 
 /**
  * Response handler for handling all the responses.
  */
+@Slf4j
 public class SmartAPIResponseHandler {
 
 	public JSONObject handle(Response response, String body) throws IOException, SmartAPIException, JSONException {
 		System.out.println("***************************");
 		if (response.header("Content-Type").contains("json")) {
 			JSONObject jsonObject = new JSONObject(body);
-			
+
 //			if (jsonObject.optString("data") == null || jsonObject.optString("data") == "") {
-			if (!jsonObject.has("status") || jsonObject.has("success")) {	
+			if (!jsonObject.has("status") || jsonObject.has("success")) {

```

Comment:
what is this condition if (!jsonObject.has("status") || jsonObject.has("success")) {?

Reply:
This condition is to check the response body.

this is the code already deployed on the main branch.at the time of formatting code, space got added
the condition was removed in code improvements PR.

Reply:
Still not sure what does this condition mean?
if (!jsonObject.has("status") || jsonObject.has("success")) {
Is this mean jsonObject is not having tag status and having success tag?  If this is true then , what is trying to achieve here ?

Reply:
This is legacy code..
yes it is checking for status and success tag.
the thing the code is trying to achieve here is on the basis of this condition, further the error code is handled.

As discussed in last call this condition will be changed in code improvement ticket.